custom/src/main/java/com/solarwinds/opentelemetry/extensions/config/DeclarativeLoader.java

```diff
@@ -0,0 +1,65 @@
+package com.solarwinds.opentelemetry.extensions.config;
+
+import com.google.auto.service.AutoService;
+import com.solarwinds.joboe.config.ConfigContainer;
+import com.solarwinds.joboe.config.ConfigGroup;
+import com.solarwinds.joboe.config.InvalidConfigException;
+import com.solarwinds.joboe.config.JavaRuntimeVersionChecker;
+import com.solarwinds.joboe.logging.Logger;
+import com.solarwinds.joboe.logging.LoggerFactory;
+import com.solarwinds.opentelemetry.extensions.LoggingConfigProvider;
+import com.solarwinds.opentelemetry.extensions.config.parser.yaml.DeclarativeConfigParser;
+import com.solarwinds.opentelemetry.extensions.provider.AutoConfigurationCustomizerProviderImpl;
+import io.opentelemetry.api.incubator.config.ConfigProvider;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.javaagent.tooling.BeforeAgentListener;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.internal.AutoConfigureUtil;
+
+@AutoService(BeforeAgentListener.class)
+public class DeclarativeLoader implements BeforeAgentListener {
+  private static final Logger logger = LoggerFactory.getLogger();
+
+  @Override
+  public void beforeAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
+    ConfigProvider configProvider =
+        AutoConfigureUtil.getConfigProvider(autoConfiguredOpenTelemetrySdk);
+
+    if (configProvider != null) {
+      boolean jdkVersionSupported = JavaRuntimeVersionChecker.isJdkVersionSupported();
+      AutoConfigurationCustomizerProviderImpl.setAgentEnabled(jdkVersionSupported);
+
+      DeclarativeConfigProperties instrumentationConfig = configProvider.getInstrumentationConfig();
+      if (instrumentationConfig != null) {
+        DeclarativeConfigProperties solarwinds =
+            instrumentationConfig
+                .getStructured("java", DeclarativeConfigProperties.empty())
+                .getStructured("solarwinds", DeclarativeConfigProperties.empty());
+
+        try {
+          ConfigContainer configContainer = new ConfigContainer();
+          new DeclarativeConfigParser(configContainer).parse(solarwinds);
+          ConfigurationLoader.processConfigs(configContainer);
+
+          ConfigContainer agentConfig = configContainer.subset(ConfigGroup.AGENT);
+          LoggerFactory.init(LoggingConfigProvider.getLoggerConfiguration(agentConfig));
+          logger.debug("Loaded via declarative config");
```

Comment:
I think the declarative config changes are critical enough that it'll be good to make this an INFO level (default emitted) message instead.